src/Compilers/CSharp/Test/Semantic/Semantics/CollectionLiteralTests.cs

```diff
@@ -348,9 +348,12 @@ static void Main()
                 """;
             var comp = CreateCompilation(source);
             comp.VerifyEmitDiagnostics(
-                // (5,13): error CS0815: Cannot assign collection literals to an implicitly-typed variable
+                // (5,17): error CS9503: There is no target type for the collection literal.
                 //         var z = [[3]];
-                Diagnostic(ErrorCode.ERR_ImplicitlyTypedVariableAssignedBadValue, "z = [[3]]").WithArguments("collection literals").WithLocation(5, 13));
+                Diagnostic(ErrorCode.ERR_CollectionLiteralNoTargetType, "[[3]]").WithLocation(5, 17),
+                // (5,18): error CS9503: There is no target type for the collection literal.
```

Comment:
I'm guessing there's no obvious place to hang a "target type is an error type, therefore I shouldn't error when converting for error recovery" check on the inner expression?

Reply:
Fixed, thanks.